sklearn/preprocessing/data.py

```diff
@@ -397,6 +399,239 @@ def inverse_transform(self, X, copy=None):
+        elif isinstance(scale, np.ndarray):
+            scale[scale == 0.0] = 1.0
+            scale[-np.isfinite(scale)] = 1.0
+        return scale
+
+    def fit(self, X, y=None):
+        """Compute the median and quantiles to be used for scaling.
+
+        Parameters
+        ----------
+        X : array-like with shape [n_samples, n_features]
+            The data used to compute the median and quantiles
+            used for later scaling along the features axis.
+        """
+        if sparse.issparse(X):
+            raise TypeError("RobustScaler cannot be fitted on sparse inputs")
+
+        X = self._check_array(X, self.copy)
+        if self.with_centering:
+            self.center_ = np.median(X, axis=0)
+
+        if self.with_scaling:
+            q = np.percentile(X, (25, 75), axis=0)
+            self.scale_ = (q[1] - q[0])
+            if np.isscalar(self.scale_):
+                if self.scale_ == 0:
+                    self.scale_ = 1.
+            else:
+                self.scale_[self.scale_ == 0.0] = 1.0
+                self.scale_[~np.isfinite(self.scale_)] = 1.0
+        return self
+
+    def transform(self, X, y=None):
+        """Center and scale the data
+
+        Parameters
+        ----------
+        X : array-like or CSR matrix.
+            The data used to scale along the specified axis.
+        """
+        if self.with_centering:
+            check_is_fitted(self, 'center_')
+        if self.with_scaling:
+            check_is_fitted(self, 'scale_')
+        X = self._check_array(X, self.copy)
+        if sparse.issparse(X):
+            if self.with_scaling:
+                if X.shape[0] == 1:
+                    inplace_row_scale(X, 1.0 / self.scale_)
+                elif self.axis == 0:
+                    inplace_column_scale(X, 1.0 / self.scale_)
+        else:
+            if self.copy:
+                X = X.copy()
+            if self.with_centering:
+                X -= self.center_
+            if self.with_scaling:
+                X /= self.scale_
+        return X
+
+    def inverse_transform(self, X):
+        """Scale back the data to the original representation
+
+        Parameters
+        ----------
+        X : array-like or CSR matrix.
+            The data used to scale along the specified axis.
+        """
+        if self.with_centering:
+            check_is_fitted(self, 'center_')
+        if self.with_scaling:
+            check_is_fitted(self, 'scale_')
+        X = self._check_array(X, self.copy)
+        if sparse.issparse(X):
+            if self.with_scaling:
+                if X.shape[0] == 1:
+                    inplace_row_scale(X, self.scale_)
+                else:
+                    inplace_column_scale(X, self.scale_)
+        else:
+            if self.copy:
+                X = X.copy()
+
+            if self.with_scaling:
+                X *= self.scale_
+            if self.with_centering:
+                X += self.center_
```

Comment:
I tried, but couldn't come up with a good factorization. I noticed 4 redundant lines (`self.copy` is checked once within `check_array` and once within the method itself), but other than that, refactoring makes the code just more complex/adds a level of indirection, but barely saves any code. There are two redundant parts:

1. I could factor out the `check_is_fitted` calls and the `_check_array`. However, that results in a total of 1 saved line, which I don't think warants including a new method.
2. Scaling the sparse matrix is the same in both methods, so that could be factorized out. But again, that would save just 1 line.

I can't combine these two redundancies into a single method, because that would mean splitting up the `if sparse.issparse` check over two methods, which makes the code necessarily hard to understand.